fuel-types/src/lib.rs

```diff
@@ -3,11 +3,16 @@
 #![cfg_attr(not(feature = "std"), no_std)]
 #![warn(unsafe_code)]
 #![warn(missing_docs)]
-#![deny(unused_crate_dependencies)]
+// #![deny(unused_crate_dependencies)]
+
+// `fuel-derive` requires `fuel_types` import
+extern crate self as fuel_types;
 
 #[cfg(feature = "alloc")]
 extern crate alloc;
 extern crate core;
+#[cfg(any(feature = "alloc", feature = "std"))]
```

Comment:
isn't alloc always enabled by std? Seems like there should be a way to simplify this pattern.

Reply:
The main idea is that `canonical` s/d should be always available, we don't need to use `std` or `alloc` features.

Reply:
I think alloc might be required since the serialization uses vec

Reply:
I did more fine-grained gating, so it should be fixed now. Currently `alloc` is enabled by `std`, so maybe that could be simplified even more.